zaptest/observer/observer.go

```diff
@@ -116,7 +123,9 @@ func (o *ObservedLogs) FilterFieldKey(key string) *ObservedLogs {
 	})
 }
 
-func (o *ObservedLogs) filter(match func(LoggedEntry) bool) *ObservedLogs {
+// Filter filters entries to those for which the provided function returns
+// true.
```

Comment:
Probably worth clarifying that this returns a new ObservedLogs.

How about,

```suggestion
// Filter returns a copy of this ObservedLogs containing only those entries
// for which the provided function returns true.
```